src/ndarray/unary_function-inl.h

```diff
@@ -102,11 +102,24 @@ struct square_root_grad {
     return 0.5f / a;
   }
 };
+
 MXNET_REGISTER_TBLOB_FUN(sqrt, XPU)
 .set_function(XPU::kDevMask, UnaryForward_<XPU, op::mshadow_op::square_root>, true)
-.set_gradient(XPU::kDevMask, UnaryBackwardUseOut_<XPU, square_root_grad>, true)
+.set_gradient(XPU::kDevMask, UnaryBackwardUseIn_<XPU, square_root_grad>, true)
 .describe("Take square root of the src");
 
+// exp
+MXNET_REGISTER_TBLOB_FUN(exp, XPU)
+.set_function(XPU::kDevMask, UnaryForward_<XPU, op::mshadow_op::exp>, true)
+.set_gradient(XPU::kDevMask, UnaryBackwardUseIn_<XPU, op::mshadow_op::exp_grad>, true)
```

Comment:
why log is different in this case ? where should I look at the difference between UseIn and UseOut more in detail?